features/cellular/framework/targets/TELIT/ME910/TELIT_ME910_CellularContext.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "TELIT_ME910_CellularContext.h"
+#include "CellularLog.h"
+
+#include "Semaphore.h"
+
+namespace mbed {
+
+TELIT_ME910_CellularContext::TELIT_ME910_CellularContext(ATHandler &at, CellularDevice *device, const char *apn, bool cp_req, bool nonip_req) :
+    AT_CellularContext(at, device, apn, cp_req, nonip_req)
+{
+}
+
+TELIT_ME910_CellularContext::~TELIT_ME910_CellularContext()
+{
+}
+
+bool TELIT_ME910_CellularContext::get_context()
+{
+    bool modem_supports_ipv6 = get_property(PROPERTY_IPV6_PDP_TYPE);
+    bool modem_supports_ipv4 = get_property(PROPERTY_IPV4_PDP_TYPE);
+    _at.cmd_start("AT+CGDCONT?");
+    _at.cmd_stop();
+    _at.resp_start("+CGDCONT:");
+    _cid = -1;
+    int cid_max = 0; // needed when creating new context
+    char apn[MAX_ACCESSPOINT_NAME_LENGTH];
+    int apn_len = 0;
+
+    while (_at.info_resp()) {
+        int cid = _at.read_int();
+        if (cid > cid_max) {
+            cid_max = cid;
+        }
+        char pdp_type_from_context[10];
+        int pdp_type_len = _at.read_string(pdp_type_from_context, sizeof(pdp_type_from_context) - 1);
+        if (pdp_type_len > 0) {
+            apn_len = _at.read_string(apn, sizeof(apn) - 1);
+            if (apn_len >= 0) {
+                if (_apn && apn_len > 0 && (strcmp(apn, _apn) != 0)) {
```

Comment:
This modified version of the `get_context()` function adds a check here to only skip over (i.e. `continue`) a candidate APN if it's string length is greater than 0 (i.e. it's not an empty string).

Original:
```c++
if (_apn && (strcmp(apn, _apn) != 0)) {
    continue;
}
```
Modified:
```c++
if (_apn && apn_len > 0 && (strcmp(apn, _apn) != 0)) {
    continue;
}
```

Reply:
You are already checking length in previous if: `if (apn_len >= 0) {`, could this be changed to accept only greater than zero? Then original would work and no double checking for length?

Reply:
So even if _apn has been defined you wan't to accept an pdp context with empty apn if `context matches by pdp type?

Reply:
@jarvte The initial motivation for this change was a scenario that I saw with this module where 6 contexts were already defined (based on the return value from the `AT+CGDCONT?` command), but all 6 had an empty-string APN. The logic of the current function would reject all 6 of the contexts (because it doesn't exactly match the target APN), and would therefore try to create a new, 7th context which would be denied by the module (it only supports 6 contexts).

I proposed changing the logic of the `get_context()` function to NOT skip over a context with an empty-string APN (i.e. ONLY skip over a context if it has a non-empty-string APN that DOESN'T match the target APN).

Sorry if I'm explaining this poorly; see PR #10442 for more info.